include/ast.hpp

```diff
@@ -95,14 +95,18 @@ struct NullStmtNode : public StmtNode {
 
 struct IfStmtNode : public StmtNode {
   IfStmtNode(std::unique_ptr<ExprNode> expr,
-             std::unique_ptr<BlockStmtNode> block)
-      : predicate{std::move(expr)}, body{std::move(block)} {}
+             std::unique_ptr<BlockStmtNode> first_body,
+             std::unique_ptr<BlockStmtNode> second_body)
+      : predicate{std::move(expr)},
+        then{std::move(first_body)},
+        or_else{std::move(second_body)} {}
```

Comment:
```suggestion
             std::unique_ptr<BlockStmtNode> then,
             std::unique_ptr<BlockStmtNode> or_else = {})
      : predicate{std::move(expr)},
        then{std::move(then)},
        or_else{std::move(or_else)} {}
```

Here are 2 suggested changes:
1. For better clarity, I propose renaming the parameters. The function signature is crucial documentation, and expressive parameter names help convey their intended use. I don't personally use the terms "first body" and "second body" to denote the "then" and "else" parts. These names might be a bit confusing.
Just a quick note to let you know: We are using the [constructor initializer list](https://en.cppreference.com/w/cpp/language/constructor), and the compiler can distinguish between data members and parameters even if they share the same name. If initialization is done in the body of the constructor, the data members would have to be accessed through the `this` pointer (e.g., `this->then`).
2. To align with a consistent style, suggest making the "else" part default to `nullptr`. This is similar to the approach taken in the `DeclNode`:
    ```c++
      DeclNode(const std::string& id, ExprType decl_type,
           std::unique_ptr<ExprNode> init = {})
      : id{id}, type{decl_type}, init{std::move(init)} {}
    ```
    With this change, the _action_ of the production rule can be updated as follows:
    ```yacc
    IF '(' expr ')' block { $$ = std::make_unique<IfStmtNode>($3, $5); }
    ```

Reply:
Thanks for pointing out! I agree that this is a better solution for improving readability. 😄